### ISSUE TITLE
Drift since provider v4.28.0 on google_sql_database_instance.settings.location_preference.secondary_zone

### ISSUE THREAD

**Issue by lra** [terraform-google-modules/terraform-google-sql-db#320]
### TL;DR

Since upgrading to `hashicorp/google` v4.28.0, a plan drift appeared that's not fixable when using mssql:
```
~ resource "google_sql_database_instance" "default" {
id                            = "XXX"
name                          = "XXX"
# (14 unchanged attributes hidden)
~settings {
# (10 unchanged attributes hidden)
~location_preference {
-secondary_zone = "us-west1-b" ->null
# (1 unchanged attribute hidden)
            }
# (4 unchanged blocks hidden)
        }
# (1 unchanged block hidden)
    }
Plan: 0 to add, 1 to change, 0 to destroy.
```

- It's not drifting when using a provider older than v4.28.0
- Applying with terraform does not resolve the drift

### Expected behavior

No drift.

### Observed behavior

Drift on `secondary_zone = "us-west1-b" ->null`

### Terraform Configuration

```hcl
module "mssql" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/mssql"
  version = "~> 11.0"
  [...]
}
```
```


### Terraform Version

```sh
Terraform v1.2.4
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v3.18.0
+ provider registry.terraform.io/confluentinc/confluent v1.0.0
+ provider registry.terraform.io/elastic/ec v0.4.1
+ provider registry.terraform.io/hashicorp/aws v4.22.0
+ provider registry.terraform.io/hashicorp/azurerm v3.13.0
+ provider registry.terraform.io/hashicorp/external v2.2.2
+ provider registry.terraform.io/hashicorp/google v4.24.0
+ provider registry.terraform.io/hashicorp/google-beta v4.28.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.12.1
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
+ provider registry.terraform.io/mongodb/mongodbatlas v1.4.2
+ provider registry.terraform.io/strongdm/sdm v2.6.4
```
```


### Additional information

_No response_

**Comment by philip-harvey** [comment]
Same issue for PostgreSQL as well

**Comment by bharathkkb** [comment]
@lra Thanks for the report. This seems like a provider issue since it is not happening with providers less than v4.28.0. 
Can you open an issue in https://github.com/hashicorp/terraform-provider-google

**Comment by philip-harvey** [comment]
I don't think this is a provider issue, this is a new feature in v4.28.0 that is not yet supported by this module.  I think this module needs to set settings.location_preference {secondary_zone = "configured secondary zone"}

**Comment by lra** [comment]
> Probably need to add an ignore_changes lifecycle to this

Yes, and we cannot do this while using this module unless we fork it.

**Comment by philip-harvey** [comment]
Understood, I was meaning that the module needs to be fixed

**Comment by kpocius** [comment]
Oddly, I can reproduce the issue with 4.27.0 and 4.26.0 and it started happening in the last few days even though we haven't updated the module or the provider.

**Comment by ev00q** [comment]
@philip-harvey wrote:
> Same issue for PostgreSQL as well. I believe the Issue only occurs for databases that have replicas.

Actually I've just observed this with a HA Postgres 14 instance **which doesn't have any replicas**, Google provider v4.31.0.

**Comment by KrustyHack** [comment]
Ay,

It would be nice, as right now Terraform is always detecting drift when I apply my stack :

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.postgresql_bits.module.postgresql.google_sql_database_instance.default will be updated in-place
  ~ resource "google_sql_database_instance" "default" {
        id                            = "xxx"
        name                          = "xxx"
        # (12 unchanged attributes hidden)

      ~ settings {
            # (10 unchanged attributes hidden)

          ~ location_preference {
              - secondary_zone = "europe-west1-d" -> null
                # (1 unchanged attribute hidden)
            }

            # (6 unchanged blocks hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

My instance has a read replica.

**Comment by philip-harvey** [comment]
This isn't actually a fix for the issue, this is a work around. The issue is still present and is an API issue.
https://issuetracker.google.com/issues/240108929

**Comment by philip-harvey** [comment]
You can still reproduce this issue even with the enhancement/work around merged.
When you add HA to an instance and don’t specify a secondary zone after a days or operations the API will start to return the current secondary zone, which is part of the root cause of the bug.  Terraform would just keep trying to set that back to null since that is the desired state, but the API doesn’t allow this to be set back to null, which is the other part of the root cause of the bug.

To recreate the issue without having to wait for this issue to self trigger:
1/ create a CloudSQL instance with HA using the Google CloudSQL module (or any module that supports setting a secondary zone)
2/ set a secondary zone (either via Terraform, GUI or Gcloud)
3/ If you added the secondary zone via Terraform then remove it
4/ Apply the Terraform, you will now have config drift.
